test/testleakautovar.cpp

```diff
@@ -1739,6 +1740,32 @@ class TestLeakAutoVar : public TestFixture {
         ASSERT_EQUALS("", errout_str());
     }
 
+    void doublefree17() { // #8109
+        check("#include <stddef.h>\n"
+              "struct my_allocator_t\n"
+              "{\n"
+              "  char buff[100];\n"
+              "};\n"
+              "my_allocator_t a;\n"
+              "void* operator new (const size_t p_size, my_allocator_t& p_allocator)\n"
+              "{\n"
+              "  return p_allocator.buff;\n"
+              "}\n"
+              "void operator delete (void* p_ptr, my_allocator_t& p_allocator)\n"
+              "{\n"
+              "  (void) p_ptr;\n"
+              "  (void) p_allocator;\n"
+              "}\n"
+              "int test1()\n"
+              "{\n"
+              "  int* x = new (a) int;\n"
+              "  int* y = new (a) int;\n"
+              "  delete (a, x);\n"
```

Comment:
The test case is also flawed (I know it's not yours).
As it is now, the code doesn't even call the overloaded `delete`:
https://godbolt.org/z/vGhnjGqM3